### ISSUE TITLE
`Normal` distribution: Gaussian policy with zero gradient of mean head

### ISSUE THREAD

**Issue by zuoxingdong** [pytorch/pytorch#10943]
I am implementing policy gradient for continuous control by using Gaussian policy. However, it seems backpropagation through log-probability of `Normal` will lead to zero gradient of the mean head. Here is a sample code to explain the scenario

```
import torch.nn as nn
from torch.distributions import Normal


mean_head = nn.Linear(3, 1)
logvar_head = nn.Linear(3, 1)

x = torch.randn(1, 3)

mean = mean_head(x)
logvar = logvar_head(x)
std = torch.exp(0.5*logvar)

d = Normal(mean, std)
a = d.rsample()
logprob = d.log_prob(a)

L = -logprob

mean_head.zero_grad()
logvar_head.zero_grad()

L.backward()

print('loss: ', L)
print('mean: ', mean)
print('std: ', std)
print('action:', a)
print('')
print('mean grad: ', mean_head.weight.grad)
print('std grad: ', logvar_head.weight.grad)
```
it gives
```
loss:  tensor([[4.1969]], grad_fn=<NegBackward>)
mean:  tensor([[0.1695]], grad_fn=<ThAddmmBackward>)
std:  tensor([[1.0948]], grad_fn=<ExpBackward>)
action: tensor([[2.9337]], grad_fn=<ThAddBackward>)

mean grad:  tensor([[0., 0., 0.]])
std grad:  tensor([[ 0.3316, -0.8036, -0.0475]])
```

**Comment by to-mi** [comment]
Could the problem be that you are sampling `a` from the model and then computing the log prob of that? The gradient will probably flow through the sampling also? (Not sure...)

**Comment by alicanb** [comment]
Yep, you need to block gradient flow by calling `.sample()` instead of `.rsample()` (or detach it before calling `.logprob(.)`)
